verilog/tools/extractor/verilog_extractor.cc

```diff
@@ -0,0 +1,332 @@
+
+    auto instantiationBase = GetChildByTag(node, verilog::NodeEnum::kInstantiationBase);
+
+    auto instantiationType = GetChildByTag(verible::SymbolCastToNode(*instantiationBase),
+                                           verilog::NodeEnum::kInstantiationType);
+
+    auto type = verible::SymbolCastToLeaf(
+            *GetFirstChildByTag(verible::SymbolCastToNode(*instantiationType),
+                                verilog::NodeEnum::kNetVariableDeclarationAssign));
+
+    Anchor typeAnchor = Anchor(type, base);
+
+    auto variableList = GetChildByTag(verible::SymbolCastToNode(*instantiationBase),
+                                      verilog::NodeEnum::kGateInstanceRegisterVariableList);
+    auto variableName = verible::SymbolCastToLeaf(
+            *GetFirstChildByTag(verible::SymbolCastToNode(*variableList),
+                                verilog::NodeEnum::kNetVariableDeclarationAssign));
+    Anchor variableNameAnchor = Anchor(variableName, base);
+
+    Block moduleInstance = Block(Type::ModuleInstance);
+    moduleInstance.AppendAnchor(typeAnchor);
+    moduleInstance.AppendAnchor(variableNameAnchor);
+
+    return moduleInstance;
+}
+
+Anchor
+ExtractModuleEnd(const verible::SyntaxTreeNode &node, const verible::SyntaxTreeNode &root, absl::string_view base) {
+    auto moduleEndKeyword = verible::SymbolCastToLeaf(
+            *GetChildByTag(node, verilog::NodeEnum::kNetVariableDeclarationAssign));
+    Anchor moduleNameAnchor = Anchor(moduleEndKeyword, base);
+    return moduleNameAnchor;
+};
+
+Anchor
+ExtractModuleHeader(const verible::SyntaxTreeNode &node, const verible::SyntaxTreeNode &root, absl::string_view base) {
+    const auto &moduleName = verible::SymbolCastToLeaf(
+            *GetChildByTag(node, verilog::NodeEnum::kNetVariableDeclarationAssign));
+    Anchor moduleNameAnchor = Anchor(moduleName, base);
+    return moduleNameAnchor;
+};
+
+Block ExtractModule(const verible::SyntaxTreeNode &node, const verible::SyntaxTreeNode &root, absl::string_view base) {
+    Block moduleBlock = Block(Type::Module);
+
+    auto moduleHeader = GetChildByTag(node, verilog::NodeEnum::kModuleHeader);
+    moduleBlock.AppendAnchor(ExtractModuleHeader(verible::SymbolCastToNode(*moduleHeader), node, base));
+
+    auto moduleItemList = GetChildByTag(node, verilog::NodeEnum::kModuleItemList);
+    auto blocks = Extract(verible::SymbolCastToNode(*moduleItemList), base);
+    for (auto block : blocks) {
+        moduleBlock.AppendChild(block);
+    }
+
+    auto moduleEnd = GetChildByTag(node, verilog::NodeEnum::kLabel);
+    if (moduleEnd != nullptr) {
+        moduleBlock.AppendAnchor(ExtractModuleEnd(verible::SymbolCastToNode(*moduleEnd), node, base));
+    }
+
+    return moduleBlock;
+};
+
+void InitializeTagResolver() {
+    tagExtractor[static_cast<int>(verilog::NodeEnum::kModuleDeclaration)] = ExtractModule;
+    tagExtractor[static_cast<int>(verilog::NodeEnum::kDataDeclaration)] = ExtractModuleInstantiation;
+}
```

Comment:
You can use initializers like:

```c++
const std::map<...> kMyMap{
  {key1, value1},
  {key2, value2},
  ...
};
```

Depending on your use pattern, this may not have to be global either.  For testing purposes, I generally recommend passing in such structures to functions that need them.